stubs/openpyxl/openpyxl/workbook/workbook.pyi

```diff
@@ -17,6 +17,22 @@ from openpyxl.worksheet.worksheet import Worksheet
 _WorkbookWorksheet: TypeAlias = Worksheet | WriteOnlyWorksheet | ReadOnlyWorksheet
 _WorkbookSheet: TypeAlias = _WorkbookWorksheet | Chartsheet
 
+# The type of worksheets in a workbook are the same as the `*Param` aliases above.
+# However, because Worksheet adds a lots of attributes that other _WorkbookChild subclasses don't have
+# (ReadOnlyWorksheet doesn't even inherit form it), this ends up being too disruptive
+# to the typical usage of openpyxl where sheets are just Worksheets.
+# Using Any may just loose too much type information and duck-typing
+# from Worksheet works great here, even if the instance type might be wrong.
+@type_check_only
+class _WorksheetLike(  # type: ignore[misc]
+    Worksheet, WriteOnlyWorksheet, ReadOnlyWorksheet, Protocol  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
A Protocol that inherits from a concrete class isn't meaningful. You can ignore away the error, but there is no guarantee about how type checkers will treat such a type. I don't think we should do this.

Reply:
In hindsight, I think that the fact this special class inherits from all possible subtypes solves what I was trying to solve without the need for a protocol.

Is has all the attributes of a Worksheet, and works with `isinstance`, or being passed to a function expecting a specific sheet type.